File: StudentGradingSystem.Api/Models/Attendance.cs
namespace StudentGradingSystem.Api.Models;

public class Attendance
{
    public int Id { get; set; }

    public int StudentId { get; set; }

    public Student Student { get; set; } = null!;

    public int SubjectId { get; set; }

    public Subject Subject { get; set; } = null!;

    public int FacultyId { get; set; }

    public Faculty Faculty { get; set; } = null!;

    public DateOnly Date { get; set; }

    public bool IsPresent { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: student-grading-system-web/src/types/attendance.ts
export interface StudentAttendance {
  studentId: number;
  isPresent: boolean;
}

export interface MarkAttendanceRequest {
  subjectId: number;
  date: string;
  students: StudentAttendance[];
}

export interface AttendanceResponse {
  studentId: number;
  studentName: string;
  isPresent: boolean;
}